### ISSUE TITLE
boxplot raises KeyError when notch == True and one of the boxes has no data.

### ISSUE THREAD

**Issue by mlovci** [matplotlib/matplotlib#4071]
`plt.boxplot([[0], [],], notch=False)` successfully returns:

```
{'boxes': [<matplotlib.lines.Line2D at 0x7fa1a1150e50>,
  <matplotlib.lines.Line2D at 0x7fa1a1271690>],
 'caps': [<matplotlib.lines.Line2D at 0x7fa1a1269990>,
  <matplotlib.lines.Line2D at 0x7fa1a1269fd0>,
  <matplotlib.lines.Line2D at 0x7fa1a12815d0>,
  <matplotlib.lines.Line2D at 0x7fa1a1281c10>],
 'fliers': [<matplotlib.lines.Line2D at 0x7fa1a125ac90>,
  <matplotlib.lines.Line2D at 0x7fa1a128d8d0>],
 'means': [],
 'medians': [<matplotlib.lines.Line2D at 0x7fa1a125a650>,
  <matplotlib.lines.Line2D at 0x7fa1a128d290>],
 'whiskers': [<matplotlib.lines.Line2D at 0x7fa1a1150610>,
  <matplotlib.lines.Line2D at 0x7fa1a1269390>,
  <matplotlib.lines.Line2D at 0x7fa1a1271910>,
  <matplotlib.lines.Line2D at 0x7fa1a1271f50>]}
```

but, `plt.boxplot([[0], [],], notch=True, )` returns this traceback:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-29-a605f59ca230> in <module>()  ----> 1 plt.boxplot([[0], [],], notch=True, )

/usr/local/lib/python2.7/dist-packages/matplotlib/pyplot.pyc in boxplot(x, notch, sym, vert, whis, positions, widths, patch_artist, bootstrap, usermedians, conf_intervals, meanline, showmeans, showcaps, showbox, showfliers, boxprops, labels, flierprops, medianprops, meanprops, capprops, whiskerprops, manage_xticks, hold)
   2640                          flierprops=flierprops, medianprops=medianprops,
   2641                          meanprops=meanprops, capprops=capprops,
-> 2642                          whiskerprops=whiskerprops, manage_xticks=manage_xticks)
   2643         draw_if_interactive()
   2644     finally:

/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_axes.pyc in boxplot(self, x, notch, sym, vert, whis, positions, widths, patch_artist, bootstrap, usermedians, conf_intervals, meanline, showmeans, showcaps, showbox, showfliers, boxprops, labels, flierprops, medianprops, meanprops, capprops, whiskerprops, manage_xticks)
   3117                            meanline=meanline, showfliers=showfliers,
   3118                            capprops=capprops, whiskerprops=whiskerprops,
-> 3119                            manage_xticks=manage_xticks)
   3120         return artists
   3121 

/usr/local/lib/python2.7/dist-packages/matplotlib/axes/_axes.pyc in bxp(self, bxpstats, positions, widths, vert, patch_artist, shownotches, showmeans, showcaps, showbox, showfliers, boxprops, whiskerprops, flierprops, medianprops, capprops, meanprops, meanline, manage_xticks)
   3429                          box_left]
   3430                 box_y = [stats['q1'], stats['q1'], stats['cilo'],
-> 3431                          stats['med'], stats['cihi'], stats['q3'],
   3432                          stats['q3'], stats['cihi'], stats['med'],
   3433                          stats['cilo'], stats['q1']]

KeyError: u'cihi'
```

In my opinion, it should exit gracefully with no data regardless of `notch == True/False`

**Comment by tacaswell** [comment]
Seems reasonable enough.  Want to take a crack at it? 

attn @phobson

**Comment by mlovci** [comment]
PR #4072

**Comment by mlovci** [comment]
Closed PR #4072 as it doesn't handle returned values correctly. Discussion in #4072 from @pelson suggests editing `lib/matplotlib/cbook.py`

**Comment by mlovci** [comment]
This is fixed in 1.5.dev1. Closing this issue because it's only a problem up to 1.4.2

**Comment by tacaswell** [comment]
Is this also fixed in 1.4.3?
